File: src/main/java/me/zeroeightsix/kami/feature/module/player/PortalGodMode.java
package me.zeroeightsix.kami.feature.module.player;

import me.zero.alpine.listener.EventHandler;
import me.zero.alpine.listener.Listener;
import me.zeroeightsix.kami.event.events.PacketEvent;
import me.zeroeightsix.kami.feature.module.Module;
import net.minecraft.network.packet.c2s.play.TeleportConfirmC2SPacket;

/**
 * Created by GlowskiBroski on 10/14/2018.
 */
@Module.Info(name = "PortalGodMode", category = Module.Category.PLAYER)
public class PortalGodMode extends Module {

    @EventHandler
    public Listener<PacketEvent.Send> listener = new Listener<>(event -> {
        if (isEnabled() && event.getPacket() instanceof TeleportConfirmC2SPacket) {
            event.cancel();
        }
    });

}

File: src/main/kotlin/me/zeroeightsix/kami/feature/module/PortalGodMode.kt
package me.zeroeightsix.kami.feature.module

import me.zero.alpine.listener.EventHandler
import me.zero.alpine.listener.Listener
import me.zeroeightsix.kami.event.PacketEvent.Send
import net.minecraft.network.packet.c2s.play.TeleportConfirmC2SPacket

@Module.Info(name = "PortalGodMode", category = Module.Category.PLAYER)
object PortalGodMode : Module() {
    var sent: TeleportConfirmC2SPacket? = null

    override fun onEnable() {
        sent = null
    }

    override fun onDisable() {
        sent?.let {
            mc.networkHandler?.sendPacket(sent)
        }
    }

    @EventHandler
    var listener = Listener({ event: Send ->
        if (event.packet is TeleportConfirmC2SPacket) {
            event.cancel()
            sent = event.packet
        }
    })
}
